go/libraries/doltcore/sqle/index/z_encoding.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2023 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package index
+
+import (
+	"bytes"
+	"math"
+	"sort"
+
+	"github.com/dolthub/go-mysql-server/sql/types"
+)
+
+// LexFloat maps the float64 into an uint64 representation in lexicographical order
+// For negative floats, we flip all the bits
+// For non-negative floats, we flip the signed bit
+func LexFloat(f float64) uint64 {
+	b := math.Float64bits(f)
+	if b>>63 == 1 {
```

Comment:
maybe just this?
```suggestion
	if b < 0 {
```

Reply:
`b` is a `uint64`, so it's never less than `0`.
It could be `f < 0.0`, but I wanted to check the sign bit.